### ISSUE TITLE
Pasting from Keyboard Maestro leads to incongruent page name creation

### ISSUE THREAD

**Issue by royYYZSFO** [Roam-Research/issues#491]
I have a KM shortcut that looks like this:

The hyphens that are meant to create bullet points in Roam, end up creating the bullet points properly, but the pages being created have a hyphen in front of the name, which is bad. This may be an issue of mine in KM, but I believe at least Roam should be consistent if it creates the Attribute as "People", not to create a page as "- People".

https://share.getcloudapp.com/7KuLbX42

**Comment by mg0716** [comment]
@royYYZSFO I agree that Roam should be smart enough to strip the prepended hyphens, but did you try the same thing in KM without the hyphens, and see if Roam bulleted/indented properly?

**Comment by royYYZSFO** [comment]
Yes - It works when you remove the hyphens. I needed to merge a number of
pages that were duplicated with a hyphen prefix, but it's solved now. I
received this from a KM template, so thought it would be handled properly
out of the box.

**Comment by Bardia95** [comment]
Sorry @royYYZSFO I could not reproduce this issue with Keyboard Maestro, they get copied in without any extra hyphens. Could you please share a video of the issue?

**Comment by royYYZSFO** [comment]
Hi Bardia - Appreciate you looking into this. Everything looks fine when
you paste, but it creates new pages for the attributes that include hyphens
(it doesn't look like it when you paste).

Let me know if that is enough detail. Thanks!

**Comment by Bardia95** [comment]
@royYYZSFO sorry but I tried reproducing it but it is working properly, can you please share a video of the issue in action?

**Comment by royYYZSFO** [comment]
Hi Bardia - Here is a simple example.

At first, there is a page for "family" but not "\tab -family"
I use the KB Maestro shortcut. In the page, it links to "family"
However, it has created a new page called "\tab -family" as well that is
tied to that page.

https://share.getcloudapp.com/bLuRdZd9

Let me know if that helps

**Comment by keenanpayne** [comment]
This happens to me as well. The example that @royYYZSFO included [here](https://share.getcloudapp.com/bLuRdZd9) showcases the problem perfectly. 

There are many different instances where using Keyboard Maestro macros to populate attributes in Roam provides non-ideal outcomes. 

For example, say I have a KM macro that pastes this text: 

```yaml
#[[my/Zettelkasten]]
  Tags:: 
  Description:: 
  Related:: 
  Sources:: 
  Notes:: 
```

When pasted by Keyboard Maestro into Roam, I receive the following attributes: 

```yaml
  Tags
  Description
  Related
  Sources
  Notes
```

The attributes will actually have two spaces before their name which makes `  Description` different from `Description`. This causes lots of duplicate attributes (i.e., pages) which is frustrating because it either takes time to clean up regularly, or you have to forego using KM to provide the convenience of auto-populating something that should be consistent. 

I've tried to use tabs in my Keyboard Maestro macro and the same problem occurred. I would get attribute names with an actual tab character prepending them. The same goes for hyphens. If I used hyphens and made what looks like a markdown list, I end up with ` - Description`. 

This all makes for a very frustrating user experience, especially because tools like KM are essential for automating workflows in Roam. 

Would love to see some help from Roam sanitizing values that get pasted into Roam (i.e., removing whitespace, using hyphens to dictate indentation rather than a character that should be included in an attribute).
